### ISSUE TITLE
Nested items in responses for Objective-C

### ISSUE THREAD

**Issue by GustavTaxen** [swagger-api/swagger-codegen#819]
I'm failing to use Swagger to generate an Objective-C client that have nested items in the "200" response for a POST operation. 

More specifically, my server responds to a certain POST operation with the following json data:

{
  "collection_of_stuff": {
    "entry_1": "some_string",
    "entry_2": "another_string",
    "entry_3": "yet_another_string"
  }
}
I describe this in Swagger like this:

        "/foo/bar": {
            "post": {
                "tags": [
                    "a_tag"
                ],
                "description": "some_description",
                "operationId": "some_id",
                "parameters": [
                    {
                        "name": "a_parameter",
                        "in": "body",
                        "description": "Some parameter",
                        "required": true,
                        "schema": {
                            "$ref": "#/definitions/ParametersIn"
                        }
                    }
                ],
                "responses": {
                    "200": {
                        "description": "Successful request",
                        "schema": {
                            "type": "object",
                            "properties": {
                                "collection_of_stuff": {
                                    "$ref": "#/definitions/Output"
                                }
                            }
                        }
                    },
                    "400": {
                        "description": "Bad request"
                    },
                    "401": {
                        "description": "Unauthorized"
                    },
                    "403": {
                        "description": "Forbidden"
                    },
                    "406": {
                        "description": "invalid Content-Type header."
                    }
                }
            }
        },
...
    "definitions": {
        "Output": {
            "type": "object",
            "properties": {
                "entry_1": {
                    "type": "string"
                },
                "entry_2": {
                    "type": "string"
                },
                "entry_3": {
                    "type": "string"
                }
            }
        }
    }

Both the online editor and codegen seems to accept this definition. But the generated Objective-C code looks like this:

    // primitive response
            // primitive response type
    return [client stringWithCompletionBlock: requestUrl 
                                              method: @"POST" 
                                         queryParams: queryParams 
                                                body: bodyDictionary 
                                        headerParams: headerParams
                                  requestContentType: requestContentType
                                 responseContentType: responseContentType
                                     completionBlock: ^(NSString *data, NSError *error) {
                         if (error) {
                             completionBlock(nil, error);
                             return;
                         }
                        NSObject *result = data ? [[NSObject  alloc]initWithString: data] : nil;
                        completionBlock(result, nil);
                     }];

In other words, the generated code doesn't parse the response at all - it tries to initialize an anonymous NSObject with the raw json string. (Which produces a compilation error.)

**Comment by wing328** [comment]
For codegen, inline schema is not supported at the moment. Please refer to #354 and #730 for more information.

To workaround the issue, you can avoid defining the inline schema by defining the model (object) in the `definitions` section

**Comment by fehguy** [comment]
Will be resolved when https://github.com/swagger-api/swagger-codegen/commit/9508579eed1346f728b395e047b06e9a323bea81 is merged to master

**Comment by fehguy** [comment]
Nested items are supported now.  Please reopen if you have any questions.
